var/spack/repos/builtin/packages/paraview/package.py

```diff
@@ -25,8 +27,8 @@ class Paraview(Package):
 
     depends_on('bzip2')
     depends_on('freetype')
-    depends_on('hdf5')
     depends_on('hdf5+mpi', when='+mpi')
+    depends_on('hdf5~mpi', when='~mpi')
```

Comment:
I'm far from being an expert in both spack as well as paraview.
As I still do not understand well variant usage in spack and how it interact with dependencies, 
and because of the -mpi setting that I had to issue on netcdf dependency in order to make the compilation possible, I would like to stay on the safe side:

My use case is to build at least two flawor of paraview: one as similar as possible as what is provided by kitware precompiled ( usable as either client or standalone for single node ), this would be feature rich ( I would like to include as many reader as possible ) stable and optimized for  HW accelerated rendering on a single node. 
Other flavours would be mainly used for scale up paraview, using pvserver using mpi and possibly with sofware rendering to run on nodes without GPU
I thought that there were no reasons to use an mpi-enabled hdf with a non mpi paraview but maybe I' m wrong.
I was trying to add variant to paraview so that a single spack install paraview  would build correct dependencies automatically.
Maybe this  is not the right way to address this problem and is better to be more generic on deps and  specify more on the install command line using ^ sintax like

spack install paraview@5.0.0+qt+python+tcl+opengl2%gcc@4.8.2 **^hdf5 -mpi ** ^netcdf -mpi 
I was not sure if this last syntax would start the build of an hdf5 without mpi if one with mpi is already available.

I can revert the mod and resubmit, if needed. 
What is the best procedure to follow? I' m quite new with github as well